### PR TITLE
Session sharing not working

### PR DESCRIPTION
**Pull request by aik099**
Session sharing wasn't working, when in same test suite there were some tests requiring `isolated` session strategy.

### CODE REVIEW

File: library/aik099/PHPUnit/Session/IsolatedSessionStrategy.php [hunk 1]
```diff
@@ -97,4 +101,16 @@ public function onTestEnd(TestEvent $event)
...
+	/**
+	 * Checks, that event can be handled by this class.
+	 *
+	 * @param TestEvent $event Test event.
+	 *
+	 * @return boolean
+	 */
+	protected function isEventForMe(TestEvent $event)
+	{
+		return get_class($event->getTestCase()->getSessionStrategy()) === get_class($this);
```
Comment (stof) [inline library/aik099/PHPUnit/Session/IsolatedSessionStrategy.php]: ``$event->getTestCase()->getSessionStrategy() instanceof self`` ?

File: library/aik099/PHPUnit/Session/IsolatedSessionStrategy.php [hunk 2]
```diff
@@ -97,4 +101,16 @@ public function onTestEnd(TestEvent $event)
...
 	}
 
+	/**
+	 * Checks, that event can be handled by this class.
+	 *
+	 * @param TestEvent $event Test event.
+	 *
+	 * @return boolean
+	 */
+	protected function isEventForMe(TestEvent $event)
```
Comment (stof) [inline library/aik099/PHPUnit/Session/IsolatedSessionStrategy.php]: what about making it private ? I don't think an extension point is worth it here

Comment (aik099) [inline library/aik099/PHPUnit/Session/IsolatedSessionStrategy.php]: Done, not pushed yet.

File: library/aik099/PHPUnit/Session/IsolatedSessionStrategy.php (hunk 1 above)
Comment (aik099) [inline library/aik099/PHPUnit/Session/IsolatedSessionStrategy.php]: Your suggestion looks awesome. Done, not pushed yet.

**Comment by aik099** [comment]
Placing a breakpoint in https://github.com/padraic/mockery/blob/master/library/Mockery/Mock.php#L704 file should help out figuring out who is doing excessive calls to `Session::stop()` and make tests pass.

**Pull request merged**